buildSrc/src/main/kotlin/com/saveourtool/save/buildutils/VersioningConfiguration.kt

```diff
@@ -23,36 +23,40 @@ import org.gradle.kotlin.dsl.getByType
  */
 fun Project.configureVersioning() {
     apply<ReckonPlugin>()
-    apply<GrgitServicePlugin>()
-    val grgitProvider = project.extensions
-        .getByType<GrgitServiceExtension>()
-        .service
-        .map { it.grgit }
 
     val isSnapshot = hasProperty("reckon.stage") && property("reckon.stage") == "snapshot"
     configure<ReckonExtension> {
-        scopeFromProp()
+        setDefaultInferredScope(Scope.MINOR.name)
+        setScopeCalc(calcScopeFromProp())
         if (isSnapshot) {
             // we should build snapshots only for snapshot publishing, so it requires explicit parameter
-            snapshotFromProp()
+            snapshots()
+            setStageCalc(calcStageFromProp())
+
+            /**
+             * A terrible hack to remove all pre-release tags. Because in semver `0.1.0-SNAPSHOT` < `0.1.0-alpha`, in snapshot mode
+             * we remove tags like `0.1.0-alpha`, and then reckoned version will still be `0.1.0-SNAPSHOT` and it will be compliant.
+             */
+            setTagParser { tagName ->
```

Comment:
may be leave it in separate function, as before?

Reply:
sure